src/main/java/dev/dirs/impl/Windows.java

```diff
@@ -2,179 +2,81 @@
 
 import dev.dirs.Constants;
 
-import java.io.BufferedReader;
-import java.io.File;
-import java.io.IOException;
-import java.io.InputStreamReader;
-import java.lang.reflect.Method;
-import java.nio.charset.Charset;
-
-public class Windows {
+import java.lang.foreign.AddressLayout;
+import java.lang.foreign.Arena;
+import java.lang.foreign.FunctionDescriptor;
+import java.lang.foreign.GroupLayout;
+import java.lang.foreign.Linker;
+import java.lang.foreign.MemoryLayout;
+import java.lang.foreign.MemorySegment;
+import java.lang.foreign.SymbolLookup;
+import java.lang.foreign.ValueLayout;
+import java.lang.invoke.MethodHandle;
+
+import static java.lang.foreign.ValueLayout.JAVA_BYTE;
+import static java.lang.foreign.ValueLayout.JAVA_CHAR;
+
+public final class Windows {
 
   private Windows() {}
 
-  private static final String UTF8_BOM = "\ufeff";
-  private static Object base64Encoder = null;
-  private static Method base64EncodeMethod = null;
-  // This string needs to end up being a multiple of 3 bytes after conversion to UTF-16. (It is currently 1200 bytes.)
-  // This is because Base64 converts 3 bytes to 4 letters; other numbers of bytes would introduce padding, which
-  // would make it harder to simply concatenate this precomputed string with whatever directories the user requests.
-  static final String SCRIPT_START_BASE64 = Constants.operatingSystem == 'w' ? toUTF16LEBase64("& {\n" +
-      "[Console]::OutputEncoding = [System.Text.Encoding]::UTF8\n" +
-      "Add-Type @\"\n" +
-      "using System;\n" +
-      "using System.Runtime.InteropServices;\n" +
-      "public class Dir {\n" +
-      "  [DllImport(\"shell32.dll\")]\n" +
-      "  private static extern int SHGetKnownFolderPath([MarshalAs(UnmanagedType.LPStruct)] Guid rfid, uint dwFlags, IntPtr hToken, out IntPtr pszPath);\n" +
-      "  public static string GetKnownFolderPath(string rfid) {\n" +
-      "    IntPtr pszPath;\n" +
-      "    if (SHGetKnownFolderPath(new Guid(rfid), 0, IntPtr.Zero, out pszPath) != 0) return \"\";\n" +
-      "    string path = Marshal.PtrToStringUni(pszPath);\n" +
-      "    Marshal.FreeCoTaskMem(pszPath);\n" +
-      "    return path;\n" +
-      "  }\n" +
-      "}\n" +
-      "\"@\n") : null;
-
-  public static String[] getWinDirs(String... guids) {
-    int guidsLength = guids.length;
-    StringBuilder buf = new StringBuilder(guidsLength * 68);
-    for (int i = 0; i < guidsLength; i++) {
-      buf.append("[Dir]::GetKnownFolderPath(\"");
-      buf.append(guids[i]);
-      buf.append("\")\n");
+  static {
+    if (Constants.operatingSystem == 'w') {
+      System.loadLibrary("combase");
+      System.loadLibrary("ole32");
+      System.loadLibrary("shell32");
```

Comment:
^combase^

Reply:
@brcolow The commit already had `combase` in it, really weird.

Reply:
Ah, I was only reading comments - should have checked the code. It's strange that it thinks the path should be inside sbt temp build directory...but maybe it is just showing the last path tried (and it doesn't show it already tried C:\Windows\System32.

Can we verify that combase.dll is present on the Microsoft Windows Server 2022 10.0.20348 CI environment? Can't imagine it isn't...we may need to ping in the [Panama](https://openjdk.org/projects/panama/) devs.

Reply:
this happens local on windows11 also, and combase.dll is present. maybe an sbt issue, i created one: https://github.com/sbt/sbt/issues/7833.

i tried to create a gradle build quickly, but run out of time before it was working.

Reply:
@FriendSeeker on https://github.com/sbt/sbt/issues/7833 suggested:

```
      String sysdir = System.getenv("WINDIR") + "/system32/";
      System.load(sysdir + "combase.dll");
      System.load(sysdir + "ole32.dll");
      System.load(sysdir + "shell32.dll");
```
which finally makes it work.

Reply:
Absolutely mysterious how everything is fine, except for `CoTaskMemFree`/`combase` ...

Reply:
@soc you want to merge it now, with the fix?

Reply:
Mhhh, which fix do you mean?

Reply:
#64 to fix this pull request.

Reply:
Ah, understood.

I think I'd rather leak a few bytes of memory than going back to concatenating strings pulled out of environment variables ...

Reply:
maybe i do not understand now :) what is the plan now to make the functionality work? the unit test still fails isn't it?

Reply:
My ideas was to drop `CoTaskMemFree(...)` and drop loading `combase`.